dcc-download-server/src/main/java/org/icgc/dcc/download/server/endpoint/ReleaseEndpoint.java

```diff
@@ -0,0 +1,97 @@
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.download.server.endpoint;
+
+import static java.lang.String.format;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.web.bind.annotation.RequestMethod.PUT;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+import org.icgc.dcc.download.server.fs.DownloadFilesReader;
+import org.icgc.dcc.download.server.service.FileSystemService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.endpoint.Endpoint;
+import org.springframework.boot.actuate.endpoint.mvc.MvcEndpoint;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@ConfigurationProperties(prefix = "endpoints." + ReleaseEndpoint.ENDPOINT_ID, ignoreUnknownFields = false)
+public class ReleaseEndpoint implements MvcEndpoint {
+
+  /**
+   * Constants.
+   */
+  protected static final String ENDPOINT_ID = "release";
+
+  @NonNull
+  private final FileSystemService fileSystemService;
+  @NonNull
+  private final DownloadFilesReader downloadFilesReader;
+
+  @RequestMapping(value = "/{releaseName}", method = PUT)
+  public ResponseEntity<String> addRelease(@PathVariable("releaseName") String releaseName) {
+    if (!downloadFilesReader.isValidReleaseName(releaseName)) {
+      log.warn("Failed to verify release '{}' against the release name regex", releaseName);
+
+      return getNotFoundResponse(format("Invalid release '%s'", releaseName));
```

Comment:
I think this solution is a simpler one, it will be used only in this controller. And secondly, I'm not sure `@ExceptionHandler` will work with `MvcEndpoint`